### PR TITLE
Major refactoring to enable nested commands

### PR DESCRIPTION
**Pull request by samuel-kerrien**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,16 +23,15 @@ start using the CLI:
 # Currently supported features
 
 ## profiles
-* adding a new profile (--add <name> --url <URL>)
-* removing a locally registered deployment (--remove <name>)
-* list locally registered deployment (no param), 
+* adding a new profile (--create <name> --url <URL>)
```
Comment (MFSY) [inline README.md]: I would make all items start with capital letter or lower case.

File: README.md [hunk 2]
```diff
@@ -23,16 +23,15 @@ start using the CLI:
...
+* Deleting a locally registered deployment (--delete <name>)
+* list locally registered deployment (--list), 
 * select a locally registered deployment (--select <name>)
 
 ## login
 * log into the currently selected local deployment of nexus
```
Comment (MFSY) [inline README.md]: Why local deployment ? Why not 'log into the currently selected deployment of Blue Brain Nexus' ?

File: tests/test_cli.py [hunk 3]
```diff
@@ -42,14 +42,14 @@ def test_nexus_contexts():
 def test_nexus_organizations():
     assert_command_returns_exist_code("nexus deployments --add foo --url https://bbp-nexus.epfl.ch/staging", 0)
     assert_command_returns_exist_code("nexus deployments --select foo", 0)
-    assert_command_returns_exist_code("nexus organizations --list", 0)
+    assert_command_returns_exist_code("nexus orgs --list", 0)
     assert_command_returns_exist_code("nexus deployments --remove foo", 0)
 
 
 def test_nexus_domains():
```
Comment (MFSY) [inline tests/test_cli.py]: MAy be projects ?

File: tests/test_cli.py [hunk 4]
```diff
@@ -42,14 +42,14 @@ def test_nexus_contexts():
 def test_nexus_organizations():
```
Comment (MFSY) [inline tests/test_cli.py]: We certainly need tests but are these tests still relevant for the new cli ?

File: tests/test_cli.py (hunk 3 above)
Comment (samuel-kerrien) [reply tests/test_cli.py]: I hadn't reworked the test, doing it now.

File: README.md (hunk 2 above)
Comment (samuel-kerrien) [reply README.md]: agreed.

File: README.md (hunk 1 above)
Comment (samuel-kerrien) [reply README.md]: done, PR update coming

File: tests/test_cli.py (hunk 4 above)
Comment (samuel-kerrien) [reply tests/test_cli.py]: nope, well spotted.

**Review by MFSY — APPROVED** [review]

**Pull request merged**